processing/src/main/java/io/druid/query/aggregation/cardinality/CardinalityAggregatorFactory.java

```diff
@@ -56,23 +88,46 @@ public static Object estimateCardinality(Object object)
   }
 
   private static final byte CACHE_TYPE_ID = (byte) 0x8;
+  private static final byte CACHE_KEY_SEPARATOR = (byte) 0xFF;
 
   private final String name;
+  private final List<DimensionSpec> fields;
   private final List<String> fieldNames;
```

Comment:
removed fieldNames, getRequiredColumns() will now generate the name list when called